### ISSUE TITLE
SPAM-X Plugin not blocking IPs using PHP7

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#771]
The preg_match call on line 178 of the file ip.examine.class.php does not work with PHP7. It works fine under PHP5.

Not sure why? @dengenxp, @mystralkk any ideas?

**Comment by mystralkk** [comment]
What pattern are you using for IP matching?

**Comment by eSilverStrike** [comment]
This is the code
```
$matches = preg_match($pattern, $ip);
```
and the incoming variables equal
```
$pattern = '#\^192\.168\.1\.224#i' 
$ip = '192.168.1.224'
```
It should work but for some reason not on PHP7 for me (fine on PHP5.6). I have been testing on IIS. 

I was testing this using forum posts initially. I tried testing using comments but the SPAM-X ip class seems to not be executing (no errors). Not sure if this is a Geeklog setting I am missing or what (I haven't had time to further test yet).

**Comment by mystralkk** [comment]
When you try to match IP v4 addresses, you don't have to think about cases.  so,

```php
$pattern = '#\^192\.168\.1\.224#' 
```
should work fine.

And there is another factor.  Look at line 177:

```PHP
177:                $pattern = $this->prepareRegularExpression($val);
```
I introduced prepareRegularExpression() method when refactoring the SPAMX plugin.  Suppose a user set the pattern to, say, just '192.168.1.224'.   Before I refactored the class, this was changed into '#192.168.1.224#' and used.  This was a right expression but not what the user intended, considering the periods were not escaped.  After refactoring, this string is changed into '#192\.168\.1\.224#' by the prepareRegularExpression() method.  See BaseCommand::prepareRegularExpression in "plugins/spamx/BaseCommand.php:122" for details.

**Comment by eSilverStrike** [comment]
Yeah I know I saw that re line 177. My initial value I entered into spam-x is 

192.168.1.224

The pattern is what line 177 returned. That is why I am trying to figure out why it is not working. Have you tested on PHP7?

**Comment by mystralkk** [comment]
I haven't tested.  I don't see why the pattern '#\^192\.168\.1\.224#i' doesn't match the value '192.168.1.224'.

**Comment by mystralkk** [comment]
Oh, I found out.  Circumflex (^) shouldn't be escaped.  Fixed with changeset 789bf3a

**Comment by eSilverStrike** [comment]
Good catch. I missed that when I went looking for the answer. The fix works.
